File: app/src/main/java/xyz/rodit/snapmod/util/UUIDUtil.java
package xyz.rodit.snapmod.util;

import java.nio.ByteBuffer;

import xyz.rodit.snapmod.mappings.UUID;

public class UUIDUtil {

    public static String fromSnap(Object snapUUID) {
        return fromByteArray((byte[]) UUID.wrap(snapUUID).getId());
    }

    public static String fromByteArray(byte[] bytes) {
        ByteBuffer bb = ByteBuffer.wrap(bytes);
        long high = bb.getLong();
        long low = bb.getLong();
        return new java.util.UUID(high, low).toString();
    }
}

File: app/src/main/java/xyz/rodit/snapmod/util/UUIDUtil.kt
package xyz.rodit.snapmod.util

import xyz.rodit.snapmod.mappings.UUID
import java.nio.ByteBuffer

fun Any.toUUIDString(): String {
    return UUID.wrap(this).toUUIDString()
}

fun UUID.toUUIDString(): String {
    return (this.id as ByteArray).toUUIDString()
}

fun ByteArray.toUUIDString(): String {
    val bb = ByteBuffer.wrap(this)
    val high = bb.long
    val low = bb.long
    return java.util.UUID(high, low).toString()
}

fun String.toSnapUUID(): UUID {
    return UUID(arrayOf(this.toUUIDBytes()))
}

fun String.toUUIDBytes(): ByteArray {
    val uuid = java.util.UUID.fromString(this)
    val bb = ByteBuffer.wrap(ByteArray(16))
    bb.putLong(uuid.mostSignificantBits)
    bb.putLong(uuid.leastSignificantBits)
    return bb.array()
}